### PR TITLE
[AC-1660] Bump SignalR Redis Package

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Bump `Microsoft.AspNetCore.SignalR.StackExchangeRedis` to the latest version that supports .NET 6. 

I used `dotnet restore` to get packages.lock.json changes.

## Code changes

* Single package update and associated lock file updates.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Notifications/Notifications.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <PropertyGroup Condition=" '$(RunConfiguration)' == 'Notifications-SelfHost' " />
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="6.0.4" />
-    <PackageReference Include="Microsoft.AspNetCore.SignalR.StackExchangeRedis" Version="6.0.4" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.StackExchangeRedis" Version="6.0.22" />
```
Comment (Hinton) [inline src/Notifications/Notifications.csproj:11]: Is there a reason redis is part of self-hosted? And do we want to bump the other SignalR dependencies, I'm not familiar with if they expect to stay in sync.

File: src/Notifications/Notifications.csproj [hunk 2]
```diff
@@ -8,7 +8,7 @@
   <PropertyGroup Condition=" '$(RunConfiguration)' == 'Notifications-SelfHost' " />
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="6.0.4" />
```
Comment (withinfocus) [inline src/Notifications/Notifications.csproj:10]: ⛏️ I would bump this at the same time.

File: src/Notifications/packages.lock.json [hunk 3]
```diff
@@ -2638,71 +2638,71 @@
...
-          "SendGrid": "9.27.0",
-          "Sentry.Serilog": "3.16.0",
-          "Serilog.AspNetCore": "5.0.0",
-          "Serilog.Extensions.Logging": "3.1.0",
-          "Serilog.Extensions.Logging.File": "2.0.0",
-          "Serilog.Sinks.AzureCosmosDB": "2.0.0",
-          "Serilog.Sinks.SyslogMessages": "2.0.6",
-          "Stripe.net": "40.0.0",
-          "YubicoDotNetClient": "1.2.0"
+          "AWSSDK.SQS": "[3.7.2.47, )",
```
Comment (withinfocus) [inline src/Notifications/packages.lock.json:2641]: ⚠️ I feel like we have to get this figured out. Different tools are doing different things, and after a bit of research I don't think we know how to control it.

File: src/Notifications/Notifications.csproj (hunk 1 above)
Comment (justindbaur) [reply src/Notifications/Notifications.csproj:11]: It's not really that it's a part of self-hosted it's just not explicitly dis-allowed. I suspect the number of SH using it is 0. 

No they don't need to be updated at the same time. This package just manages the connection lifetime and the other package is just for how we encode our messages.

File: src/Notifications/Notifications.csproj (hunk 2 above)
Comment (justindbaur) [reply src/Notifications/Notifications.csproj:10]: I disagree for something that we want to pick this late into testing, they don't have dependencies on each other and they both interact with separate parts of the SignalR stack. I'm fine with making a separate PR that only goes into `master` though. Would that be okay?

File: src/Notifications/packages.lock.json (hunk 3 above)
Comment (justindbaur) [reply src/Notifications/packages.lock.json:2641]: Yeah I don't know... that's why I made a note of the tool I used to do this one so maybe we can track it a little.

Comment (Hinton) [reply src/Notifications/packages.lock.json:2641]: Rider seems to prefer the old syntax. The dotnet CLI prefers `[)`. I think we should settle on `[)` and ask developers to run `dotnet restore -f --force-evaluate --no-cache` if they change things in Rider.

**Review by withinfocus — APPROVED** [review]
Short of the conflicting lockfile output, works for me. I think a PR to replace all the lockfiles that just goes to `master` would be nice to do after this.

**Pull request merged**